src/scripts/benefit-indicator.ts

```diff
@@ -0,0 +1,402 @@
+import parse from "parse-typed-args";
+import _ from "lodash";
+import csvToJson from "csvtojson";
+import { createObjectCsvWriter } from "csv-writer";
+
+import { D2Api, DataValueSetsDataValue } from "../types/d2-api";
+import { getApp } from "./common";
+import { getId } from "../utils/dhis2";
+import { promiseMap } from "../migrations/utils";
+
+const DE_MINISTRY_VISIT_ID = "WUMjtbgofs2";
+const DE_GROUP_ID = "i8qug2Himby";
+const DEFAULT_COC_ID = "HllvX50cXC0";
+const TARGET_COC_ID = "I8cbuxMTjjs";
+const ACTUAL_COC_ID = "oJXO2VVYWZa";
+const NEW_COC_ID = "Wj7wjri7dUs";
+const RETURNING_COC_ID = "bALUpFAqhLq";
+
+async function main() {
+    const parser = parse({
+        opts: {
+            url: {},
+            filePath: {},
+            csvPath: {},
+        },
+    });
+    const { opts } = parser(process.argv);
+
+    if (!opts.url) return;
+    if (!opts.filePath) return;
+    if (!opts.csvPath) return;
+
+    const { api } = await getApp({ baseUrl: opts.url });
+    const metadata = await getMetadata(api);
+    const dataValuesFromCsv = await readCsv(opts.filePath);
+    const { orgUnitCodes, periods } = extractOrgUnitsAndPeriods(dataValuesFromCsv);
+
+    console.debug("Fetching Org Units...");
+    const orgUnits = await getOrgUnitFromCodes(api, orgUnitCodes);
+    const orgUnitIds = orgUnits.map(getId);
+
+    console.debug("Fetching DataValues...");
+    const dataValues = await getDataValues(api, orgUnitIds, periods);
+    const onlyMinistryVisitDv = dataValues.filter(
+        dv =>
+            dv.dataElementId === DE_MINISTRY_VISIT_ID &&
+            (dv.categoryOptionComboId === NEW_COC_ID ||
+                dv.categoryOptionComboId === RETURNING_COC_ID)
+    );
+
+    const ministryDe = metadata.dataElements.find(de => de.id === DE_MINISTRY_VISIT_ID);
+    if (!ministryDe) throw Error(`Cannot found dataElement ${DE_MINISTRY_VISIT_ID}`);
+    console.debug(`${onlyMinistryVisitDv.length} dataValues for dataElement: ${ministryDe.code}`);
+
+    const dataValuesToSave = generateDataValuesToSave(
+        onlyMinistryVisitDv,
+        DEFAULT_COC_ID,
+        TARGET_COC_ID,
+        ACTUAL_COC_ID
+    );
+
+    const dvToDelete = dataValuesToSave.filter(dv => dv.deleted);
+    console.debug(`${dvToDelete.length} dataValues marked as deleted`);
+    console.debug(`${dataValuesToSave.length - dvToDelete.length} dataValues to import`);
+
+    console.debug("Saving dataValues...");
+    const resultStats = await saveDataValues(dataValuesToSave, api);
+
+    const stats = combineStats(resultStats);
+
+    console.log("Stats", JSON.stringify(stats, null, 4));
+
+    await generateReport(dataValuesToSave, orgUnits, metadata, opts.csvPath);
+    console.debug(`Report generated: ${opts.csvPath}`);
+}
+
+async function readCsv(filePath: string): Promise<DataValueCsv[]> {
+    const json = await csvToJson({
+        headers: ["orgUnit", "period", "dataElement", "aocId", "cocId", "value"],
+    }).fromFile(filePath);
+    return json;
```

Comment:
Sorry for the confusion, the CSV was intended for the client as a report, not to be used as the script, as we don't to change this specific data values, we want all the data values of that data element. Anyway, let's keep it as is, only as a note, we need to generate the CSV at the moment the script is run, to make sure we are updating everything:

```
{ echo "orgUnit,period"; curl -u "$IP_ADMIN_AUTH" 'https://platform.samaritanspurse.org/api/dataValueSets.json?orgUnit=AGZEUf9meZ6&children=true&startDate=1950&endDate=2050&dataElementGroup=ARzxqc0lEmb' | jq '.dataValues[] | [.orgUnit, .period] | @csv' -r; } > in.csv
```

And change orgUnit filter from code to id.